cli/command/node/cmd.go

```diff
@@ -38,6 +41,16 @@ func Reference(ctx context.Context, client apiclient.APIClient, ref string) (str
 		if err != nil {
 			return "", err
 		}
+		if info.Swarm.NodeID == "" {
+			// If there's no node ID in /info, the node probably
+			// isn't a manager. Call a swarm-specific endpoint to
+			// get a more specific error message.
+			_, err = client.NodeList(ctx, types.NodeListOptions{})
```

Comment:
I don't believe that combination is possible. If the node is locked, the control socket is not available.